data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwardingProcessorDecorator.java

```diff
@@ -6,26 +6,41 @@
 package org.opensearch.dataprepper.peerforwarder;
 
 import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.peerforwarder.RequiresPeerForwarding;
 import com.amazon.dataprepper.model.processor.Processor;
 import com.amazon.dataprepper.model.record.Record;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.Set;
 
 public class PeerForwardingProcessorDecorator implements Processor<Record<Event>, Record<Event>> {
     private static final Logger LOG = LoggerFactory.getLogger(PeerForwardingProcessorDecorator.class);
 
     private final Processor innerProcessor;
+    private final PeerForwarder peerForwarder;
 
-    public PeerForwardingProcessorDecorator(final Processor innerProcessor) {
+    public PeerForwardingProcessorDecorator(final Processor innerProcessor,
+                                            final PeerForwarder peerForwarder
+                                            ) {
         this.innerProcessor = innerProcessor;
+        this.peerForwarder = peerForwarder;
         LOG.info("Peer Forwarder not implemented yet, processing events locally.");
     }
 
     @Override
     public Collection<Record<Event>> execute(final Collection<Record<Event>> records) {
-        return innerProcessor.execute(records);
+        Set<String> identificationKeys = null;
+
+        if (innerProcessor instanceof RequiresPeerForwarding) {
```

Comment:
We should try to avoid this redundant check. One way is to make an interface that extends Processor and RequiresPeerForwarding.

Reply:
I like this check. But it should be in the constructor. Then you can cast the `innerProcessor` into a `RequiresPeerForwarding`, get the `identificationKeys` and keep those as a field in the class.

You can also make the extra interface like @chenqi0805 suggested. But, that would need to be another decorator and might be more complicated than necessary.

Reply:
Please also add a unit test for this check.